lambdas/pkgselect/test/test_pkgselect.py

```diff
@@ -393,3 +398,72 @@ def test_anon_access(self):
         s3_stubber.deactivate()
         client_patch.stop()
         env_patcher.stop()
+
+    def test_non_string_keys(self):
+        """
+        End-to-end test (folder view without a prefix)
+        """
+        bucket = "bucket"
+        key = ".quilt/packages/manifest_hash"
+        params = dict(
+            bucket=bucket,
+            manifest=key,
+            access_key="TESTKEY",
+            secret_key="TESTSECRET",
+            session_token="TESTSESSION"
+        )
+
+        expected_args = {
+            'Bucket': bucket,
+            'Key': key,
+            'Expression': "SELECT SUBSTRING(s.logical_key, 1) AS logical_key FROM s3object s",
+            'ExpressionType': 'SQL',
+            'InputSerialization': {
+                'CompressionType': 'NONE',
+                'JSON': {'Type': 'LINES'}
+                },
+            'OutputSerialization': {'JSON': {'RecordDelimiter': '\n'}},
+        }
+
+        # Return a response with keys that are not strings (integers here)
+        # The important test case is where all members of a column are
+        # non-string
+        logical_keys = [
+            "1",
+            "2",
+            "3",
+        ]
+        entries = []
+        for key in logical_keys:
+            entry = dict(
+                logical_key=key,
+                physical_key=key,
+                size=100
+            )
+            entries.append(json.dumps(entry))
+        jsonl = "\n".join(entries)
+        streambytes = jsonl.encode()
+        non_string_s3response = self.make_s3response(streambytes)
+
+        mock_s3 = boto3.client('s3')
+        client_patch = patch.object(
+            mock_s3,
+            'select_object_content',
+            side_effect=[
+                non_string_s3response,
+                self.s3response_meta
+            ]
+        )
+        client_patch.start()
```

Comment:
Either use `client_patch` as context manager or use `self.addCleanup(client_patch.stop)`, otherwise `client_patch.stop()` won't be called if some of asserts fail and this could make other tests fail (or maybe even give a false positive?).

Reply:
Good catch.

Reply:
It looks like you did it in the old test, but not in the new one.